src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/legalrep/SubmitClaimTest.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import uk.gov.hmcts.cmc.claimstore.idam.models.User;
+import uk.gov.hmcts.cmc.claimstore.tests.BaseLegalRepTest;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimData;
+import uk.gov.hmcts.cmc.domain.models.TimelineEvent;
+import uk.gov.hmcts.cmc.domain.models.evidence.EvidenceRow;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleEvidence;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleTimeline;
+import uk.gov.hmcts.cmc.domain.utils.LocalDateTimeFactory;
+
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+public class SubmitClaimTest extends BaseLegalRepTest {
+
+    private User solicitor;
+
+    @Before
+    public void before() {
+        solicitor = idamTestService.createSolicitor();
+    }
+
+    @Test
+    public void shouldSuccessfullySubmitClaimDataAndReturnCreatedCase() {
+        ClaimData claimData = testData.submittedBySolicitorBuilder().build();
+
+        Claim createdCase = submitClaim(claimData).then()
+                                                  .statusCode(HttpStatus.OK.value())
+                                                  .and()
+                                                  .extract()
+                                                  .body()
+                                                  .as(Claim.class);
+
+        assertThat(claimData).isEqualTo(createdCase.getClaimData());
+        assertThat(createdCase.getCreatedAt()).isCloseTo(LocalDateTimeFactory.nowInLocalZone(),
+                                                         within(2, ChronoUnit.MINUTES));
+    }
+
+    @Test
+    public void shouldReturnUnprocessableEntityWhenInvalidClaimIsSubmitted() {
+        ClaimData invalidClaimData = testData.submittedBySolicitorBuilder().withAmount(null).build();
+
+        submitClaim(invalidClaimData).then().statusCode(HttpStatus.UNPROCESSABLE_ENTITY.value());
+    }
+
+    @Test
+    public void shouldReturnConflictResponseWhenClaimDataWithDuplicatedExternalIdIsSubmitted() {
+        UUID externalId = UUID.randomUUID();
+
+        ClaimData claimData = testData.submittedBySolicitorBuilder().withExternalId(externalId).build();
+
+        submitClaim(claimData).andReturn();
+        submitClaim(claimData).then().statusCode(HttpStatus.CONFLICT.value());
+    }
+
+    @Test
+    public void shouldReturnUnprocessableEntityWhenClaimWithInvalidTimelineIsSubmitted() {
+        ClaimData invalidClaimData = testData.submittedBySolicitorBuilder()
+                                             .withTimeline(SampleTimeline.builder()
+                                                                         .withEvents(asList(new TimelineEvent[1001]))
```

Comment:
are you sure this will look like this when you press option+cmd+L (auto-formatting)?